From 2d142c09022d1b03479b7e22dd5d49a5573484ea Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Fri, 17 Mar 2017 13:29:20 -0700
Subject: [PATCH] CNS-198 Peer version failure neglects to call callback; leaks
 redis handle Reviewed by: Richard Kiene <richard.kiene@joyent.com>

---
 lib/dns-server.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/dns-server.js b/lib/dns-server.js
index 72e790d..fca411d 100644
--- a/lib/dns-server.js
+++ b/lib/dns-server.js
@@ -1083,6 +1083,7 @@ DNSServer.prototype.handleTransfer = function (q, z, cb) {
 						log.debug(perr,
 						    'unable to detect peer ' +
 						    'version');
+						ccb(perr);
 						return;
 					}
 					_.redis.set('peer:' +
-- 
2.21.0

